presto-verifier/src/test/java/com/facebook/presto/verifier/rewrite/TestQueryRewriter.java

```diff
@@ -294,6 +298,160 @@ public void testRewriteNonStorableStructuredTypes()
                         "    CAST(ROW (NULL) AS ROW(BIGINT))");
     }
 
+    @Test
+    public void testConstructMakeFunctionCallSubstituteMap()
+    {
+        assertEqualsDeep(FunctionCallRewriter.constructFunctionCallSubstituteMap("/func1(a,b)/func2(b)/"),
+                ImmutableMap.builder().put(QualifiedName.of("func1"),
+                        new FunctionCallRewriter.FunctionCallSubstitute(QualifiedName.of("func2"), ImmutableList.of(1))).build());
+        assertEqualsDeep(FunctionCallRewriter.constructFunctionCallSubstituteMap("/func1(,,1c)/func2(1c)/"),
+                ImmutableMap.builder().put(QualifiedName.of("func1"),
+                        new FunctionCallRewriter.FunctionCallSubstitute(QualifiedName.of("func2"), ImmutableList.of(2))).build());
+        assertEqualsDeep(FunctionCallRewriter.constructFunctionCallSubstituteMap("/func1(a,_,b)/func2(_)/"),
+                ImmutableMap.builder().put(QualifiedName.of("func1"),
+                        new FunctionCallRewriter.FunctionCallSubstitute(QualifiedName.of("func2"), ImmutableList.of(1))).build());
+        assertEqualsDeep(FunctionCallRewriter.constructFunctionCallSubstituteMap("/func1(a,b)/func2(b)/,/func3(a,)/func4(a)/"),
+                ImmutableMap.builder().put(QualifiedName.of("func1"),
+                        new FunctionCallRewriter.FunctionCallSubstitute(QualifiedName.of("func2"), ImmutableList.of(1))).put(QualifiedName.of("func3"),
+                        new FunctionCallRewriter.FunctionCallSubstitute(QualifiedName.of("func4"), ImmutableList.of(0))).build());
+
+        assertTrue(FunctionCallRewriter.constructFunctionCallSubstituteMap("").isEmpty());
+        assertTrue(FunctionCallRewriter.constructFunctionCallSubstituteMap("/func1(a,b)//").isEmpty());
+    }
+
+    @Test
+    public void testRewriteFunctionCalls()
+    {
+        QueryRewriter queryRewriter = getQueryRewriter(Optional.of("/approx_distinct(x)/count(x)/," +
+                "/approx_percentile(x,_)/avg(x)/," +
+                "/arbitrary(x)/min(x)/," +
+                "/map_agg(x,)/max(x)/," +
+                "/first_value(x)/min(x)/," +
+                "/map_union(x)/max(x)/," +
+                "/array_agg(x)/max(x)/"));
+
+        // Test rewriting window functions
+        assertCreateTableAs(
```

Comment:
@mbasmanova had made a comment to test some failure scenarios like where the substitute is invalid. Where are those to see that the rewrite is to blame ?

Reply:
For invalid substitution, I include the unit test testConstructMakeFunctionCallSubstituteMap() in this PR to validate the illegal substitution specification string, ex., /func1(a,b)//.

For validating illegal function signature or type signature, like /map_agg(x,)/max(x)/, I will handle this in https://github.com/prestodb/presto/pull/18095.
(The best I figured we can do in Verifier without access to HiveMetadata is to actually run the query after rewritten, fail and provide meaningful error info.)